Formal thread argument for detachVMThread method

amicic opened the pull request:
detachVMThread is a static method and can't use class member _vmThread.
It should be using thread passed through formal arguments

Fixes compile problems introduced in #8448 

Signed-off-by: Aleksandar Micic <amicic@ca.ibm.com>

amicic commented:
Jenkins test compile all jdk11

amicic commented:
Jenkins test sanity win jdk11

pshipton commented:
@amicic @AdamBrousseau what happened with the PR testing here? I see `test sanity win jdk11` however the PR build is building all platforms.

pshipton commented:
Perhaps a comment was deleted? If so, deleting it didn't stop the PR build.

pshipton commented:
PR builds running
https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/2593/ - all platforms
https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/2595/ - just windows

AdamBrousseau commented:
First one is a compile only

amicic commented:
I deleted it. After a minute or so not launching anything, I thought I mistyped something and removed and just started win sanity only. I know see that compile is on, with obviously some delay.

amicic commented:
Jenkins test compile all jdk8

pshipton commented:
The server got restarted, re-running the PR testing.

jenkins compile all jdk8

amicic commented:
I also ran IBM internal compile tests and all was fine. Ready for review/merge.
